loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+        else:
+            content = self.format_line("<", name, ">")
+
+        # disregard all quotes to ensure nice graphviz label behaviour
+        return content.replace('"', "")
+
+    def format_line(self, *items, comment=None, no_wrap=False):
+        """
+        Format a line by concatenating all items and applying indentation while observing
+        the allowed line width limit.
+
+        Note that the provided comment will simply be extended to the line and no line
+        width limit will be enforced for that.
+
+        :param list items: the items to be put on that line.
+        :param str comment: an optional inline comment to be put at the end of the line.
+        :param bool no_wrap: disable line wrapping.
+
+        :return: the string of the current line, potentially including line breaks if
+                 required to observe the line width limit.
+        :rtype: str
+        """
+
+        if no_wrap:
+            # Simply concatenate items and extend the comment
+            line = "".join(str(item) for item in items)
+        else:
+            # Use join_items to concatenate items
+            line = str(self.join_items(items, sep=""))
+        if comment:
+            return line + comment
+        return line
+
+    def visit_all(self, item, *args, **kwargs):
+        """
+        Convenience function to call :meth:`visit` for all given arguments.
+
+        If only a single argument is given that is iterable,
+        :meth:`visit` is called on all of its elements instead.
+        """
+        if is_iterable(item) and not args:
+            return list(
+                chain.from_iterable(
+                    as_tuple(self.visit(i, **kwargs) for i in item if i is not None)
+                )
+            )
+        return list(
+            chain.from_iterable(
+                as_tuple(
+                    self.visit(i, **kwargs) for i in [item, *args] if i is not None
+                )
+            )
+        )
+
+    def __add_node(self, node, **kwargs):
+        """
+        Adds a node to the graphical representation of the IR. Utilizes the
+        formatting provided by :meth:`format_node`.
+
+        Parameters
+        ----------
+        node: :any: `Node` object
+        kwargs["shape"]: str, optional (default: "oval")
+        kwargs["label"]: str, optional (default: format_node(repr(node)))
+        kwargs["parent"]: :any: `Node` object, optional (default: None)
+            If not available no edge is drawn.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            A list of a tuple of a node and potentially a edge information
+        """
+        label = kwargs.get("label", "")
+        if label == "":
+            label = self.format_node(repr(node))
+
+        shape = kwargs.get("shape", "oval")
+        label = (
+            label + "‎"
+        )  # dirty hack to force graphviz to utilize parenthesis around the label
```

Comment:
I don't understand this. Appending an empty string to `label` should not change the string itself - if that makes a difference, I suspect the root cause is something else and the add operator triggers a type conversion somehow.

Reply:
Fix with a9cb30cb65c4348747ff73a0d286fa2b89259322

Reply:
Nice find!